pkg/distributor/http.go

```diff
@@ -23,7 +23,21 @@ func (d *Distributor) PushHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (d *Distributor) OTLPPushHandler(w http.ResponseWriter, r *http.Request) {
-	d.pushHandler(w, r, push.ParseOTLPRequest)
+	interceptor := &OtelErrorHeaderInterceptor{ResponseWriter: w}
+
+	d.pushHandler(interceptor, r, push.ParseOTLPRequest)
+}
+
+type OtelErrorHeaderInterceptor struct {
+	http.ResponseWriter
+}
+
+func (i *OtelErrorHeaderInterceptor) WriteHeader(statusCode int) {
+	if statusCode == http.StatusInternalServerError {
+		i.ResponseWriter.WriteHeader(http.StatusServiceUnavailable)
+	} else {
+		i.ResponseWriter.WriteHeader(statusCode)
+	}
```

Comment:
Can you add a comment why we are doing this, e.g. link the OTLP spec https://opentelemetry.io/docs/specs/otlp/#failures-1